java.lang.reflect.InvocationTargetException

### Type of JetBrains IDE (IntelliJ, PyCharm, etc.) and version
PyCharm 2021.3.2 (build PC-213.6777.50)

### Operating System 
Windows 10 10.0 (amd64)

### TeXiFy IDEA version
0.7.16

### Description



### Stacktrace
```
java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.intellij.execution.process.ProcessHandler$2.invoke(ProcessHandler.java:247)
	at com.sun.proxy.$Proxy46.processTerminated(Unknown Source)
	at com.intellij.execution.process.ProcessHandler.lambda$notifyTerminated$2(ProcessHandler.java:205)
	at com.intellij.execution.process.ProcessHandler$TasksRunner.execute(ProcessHandler.java:280)
	at com.intellij.execution.process.ProcessHandler.notifyTerminated(ProcessHandler.java:188)
	at com.intellij.execution.process.ProcessHandler.notifyProcessTerminated(ProcessHandler.java:184)
	at com.intellij.execution.process.KillableProcessHandler.notifyProcessTerminated(KillableProcessHandler.java:148)
	at com.intellij.execution.process.BaseProcessHandler.onOSProcessTerminated(BaseProcessHandler.java:61)
	at com.intellij.execution.process.OSProcessHandler.onOSProcessTerminated(OSProcessHandler.java:196)
	at com.intellij.execution.process.BaseOSProcessHandler$1.lambda$startNotified$0(BaseOSProcessHandler.java:98)
	at com.intellij.execution.process.ProcessWaitFor.lambda$new$0(ProcessWaitFor.java:45)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:213)
	at com.intellij.execution.process.ProcessWaitFor.lambda$new$1(ProcessWaitFor.java:23)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:668)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:665)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:665)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: com.intellij.execution.ExecutionException: Main file not found
	at nl.hannahsten.texifyidea.run.latex.MakeindexUtilKt$getMakeindexOptions$$inlined$runReadAction$1.compute(actions.kt:60)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:852)
	at nl.hannahsten.texifyidea.run.latex.MakeindexUtilKt.getMakeindexOptions(MakeindexUtil.kt:94)
	at nl.hannahsten.texifyidea.run.makeindex.RunMakeindexListener.processTerminated(RunMakeindexListener.kt:47)
	... 26 more
```

--- a/src/nl/hannahsten/texifyidea/run/latex/MakeindexUtil.kt
+++ b/src/nl/hannahsten/texifyidea/run/latex/MakeindexUtil.kt
@@ -1,6 +1,8 @@
 package nl.hannahsten.texifyidea.run.latex
 
 import com.intellij.execution.ExecutionException
+import com.intellij.notification.Notification
+import com.intellij.notification.NotificationType
 import com.intellij.openapi.application.runReadAction
 import com.intellij.openapi.project.Project
 import com.intellij.openapi.vfs.VirtualFile
@@ -77,9 +79,14 @@ private fun getIndexPackageOptions(mainFile: VirtualFile?, project: Project): Li
 /**
  * Get optional parameters of the \makeindex command. If an option key does not have a value it will map to the empty string.
  */
-fun getMakeindexOptions(mainFile: VirtualFile?, project: Project): HashMap<String, String> {
+fun getMakeindexOptions(mainFile: VirtualFile?, project: Project): Map<String, String> {
     return runReadAction {
-        val mainPsiFile = mainFile?.psiFile(project) ?: throw ExecutionException("Main file not found")
+        val mainPsiFile = mainFile?.psiFile(project)
+        if (mainPsiFile == null) {
+            Notification("LaTeX", "Could not find main file ${mainFile?.path}", "Please make sure the main file exists.", NotificationType.ERROR).notify(project)
+            return@runReadAction mapOf<String, String>()
+        }
+
         val makeindexOptions = HashMap<String, String>()
         LatexCommandsIndex.getItemsInFileSet(mainPsiFile)
             .filter { it.commandToken.text == "\\makeindex" }

--- a/src/nl/hannahsten/texifyidea/run/makeindex/MakeindexCommandLineState.kt
+++ b/src/nl/hannahsten/texifyidea/run/makeindex/MakeindexCommandLineState.kt
@@ -18,7 +18,7 @@ class MakeindexCommandLineState(
     environment: ExecutionEnvironment,
     private val mainFile: VirtualFile?,
     private val workingDirectory: VirtualFile?,
-    private val makeindexOptions: HashMap<String, String>,
+    private val makeindexOptions: Map<String, String>,
     private val indexProgram: MakeindexProgram
 ) : CommandLineState(environment) {
 

